clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -406,6 +406,10 @@ public void close() {
         } finally {
             super.close();
         }
+        Node coordinator;
+        while ((coordinator = coordinator()) != null && client.pendingRequestCount(coordinator) > 0) {
```

Comment:
@hachikuji Thank you for the review. Yes, I was in two minds about adding a hard-coded timeout. But it does make sense to have a timeout, have added a timeout of 5 seconds.